articles/101_build_tool.md

```diff
@@ -289,59 +291,142 @@ The following items are possible extension points to provide custom functionalit
+
+- The jobs and scripts on *ci.ros2.org* need to be updated to invoke the universal build tool instead of `ament_tools`.
+- The `ros_buildfarm` package needs to be updated to invoke the universal build tool instead of `catkin_make_isolated`.
+  The ROS 2 buildfarm would use this modification for the upcoming ROS 2 *Bouncy* release.
+  The ROS 1 buildfarm could use the same modification in the future.
+
+For option **A)** the follow items would need to be addressed:
+
+- Support for setup files generated by `ament_cmake`.
+- Support additional packages types: plain Python packages, CMake packages without a manifest.
+- Support for Windows using `.bat` files.
+- Support for the package manifest format version 3.
+
+For option **B)** the follow items would need to be addressed.
+
+- Address user feedback when the tool is being used by a broader audience.
+
+### Future
+
+The long term goal is that the universal build tool will be used in ROS 1, in ROS 2 as well as other non-ROS projects.
+There is currently no time line when the tool will be used on the ROS 1 build or be recommended to ROS 1 users.
+This solely depends on the resources available for ROS 1.
+
+Beside that for both options there is follow up work beyond the immediate goals.
+The following enumerates a few of them but is by no means exhaustive:
+
+For option **A)** the follow items should be considered:
+
+- Support for Python packages using a `setup.cfg` file.
+- Support for `PowerShell` to work around length limitations for environment variable on Windows.
+- Support to pass package specific argument.
+- Remove support for the *devel space* concept in ROS 1.
+- Update code base to Python 3.5+.
+- Refactor code base to reduce coupling (e.g. separate [API](https://github.com/catkin/catkin_tools/blob/2cae17f8f32b0193384d2c7734afee1c60c4add2/catkin_tools/execution/controllers.py#L183-L205) for output handling).
+- Additional functionality to build Gazebo including its dependencies.
+
+For option **B)** the follow items should be considered:
 
-- Refactor the software architecture of the existing code base to support the flexibility sketched by the extension points listed above.
+- Support cross compilation.
+- Support `DESTDIR`.
+- Support a feature similar to the `profile` verb of `catkin_tools`.
 
-- Move `catkin` specific concepts out of the core of the build tool into a catkin specific extension (e.g. manifest format, *devel space*).
+## Summary and Decision
 
-- Support for ROS 2 which includes:
+Based on the above information a decision has been made to pick `colcon` as the universal build tool which was not an easy one.
 
-  - Support for Python 3 and Windows.
-  - Support for pure Python packages as well as packages without an in-source manifest file.
-  - Environment setup of `ament` packages
+To elaborate on the rational one significant advantage of `colcon` is that it is ready to be deployed for ROS 2 right now and it covers our current use cases.
```

Comment:
... on the rational**e** ...

Reply:
Fixed in 2e9804e00fc8775f8c10d80cb5f647f3533078b9.